packages/feedback/src/integration.ts

```diff
@@ -0,0 +1,323 @@
+   */
+  public attachTo(el: Element | string, optionOverrides: OptionalFeedbackConfiguration): Widget | null {
+    try {
+      const options = mergeOptions(this.options, optionOverrides);
+
+      return this._ensureShadowHost<Widget | null>(options, ({ shadow }) => {
+        const targetEl =
+          typeof el === 'string' ? doc.querySelector(el) : typeof el.addEventListener === 'function' ? el : null;
+
+        if (!targetEl) {
+          logger.error('[Feedback] Unable to attach to target element');
+          return null;
+        }
+
+        const widget = createWidget({ shadow, options, attachTo: targetEl });
+        this._widgets.add(widget);
+        return widget;
+      });
+    } catch (err) {
+      logger.error(err);
+      return null;
+    }
+  }
+
+  /**
+   * Creates a new widget. Accepts partial options to override any options passed to constructor.
+   */
+  public createWidget(optionOverrides: OptionalFeedbackConfiguration): Widget | null {
+    try {
+      return this._createWidget(mergeOptions(this.options, optionOverrides));
+    } catch (err) {
+      logger.error(err);
+      return null;
+    }
+  }
+
+  /**
+   * Removes a single widget
+   */
+  public removeWidget(widget: Widget | null | undefined): boolean {
+    if (!widget) {
+      return false;
+    }
+
+    try {
+      if (this._widgets.has(widget)) {
+        widget.removeActor();
+        widget.removeDialog();
+        this._widgets.delete(widget);
+        return true;
+      }
+    } catch (err) {
+      logger.error(err);
+    }
+
+    return false;
+  }
+
+  /**
+   * Removes the Feedback integration (including host, shadow DOM, and all widgets)
+   */
+  public remove(): void {
+    if (this._host) {
+      this._host.remove();
+    }
+    this._initialize();
+  }
+
+  /**
+   * Initializes values of protected properties
+   */
+  protected _initialize(): void {
+    this._host = null;
+    this._shadow = null;
+    this._widget = null;
+    this._widgets = new Set();
+    this._hasInsertedActorStyles = false;
+  }
+
+  /**
+   * Creates a new widget, after ensuring shadow DOM exists
+   */
+  protected _createWidget(options: FeedbackInternalOptions): Widget | null {
+    return this._ensureShadowHost<Widget>(options, ({ shadow }) => {
+      const widget = createWidget({ shadow, options });
+
+      if (!this._hasInsertedActorStyles && widget.actor) {
+        shadow.appendChild(createActorStyles(doc));
+        this._hasInsertedActorStyles = true;
+      }
+
+      this._widgets.add(widget);
+      return widget;
+    });
+  }
+
+  /**
+   * Ensures that shadow DOM exists and is added to the DOM
+   */
+  protected _ensureShadowHost<T>(
+    options: FeedbackInternalOptions,
+    cb: (createShadowHostResult: ReturnType<typeof createShadowHost>) => T,
+  ): T | null {
+    let needsAppendHost = false;
+
+    // Don't create if it already exists
+    if (!this._shadow || !this._host) {
+      const { shadow, host } = createShadowHost({
```

Comment:
l: this may be slightly easier to read like this:

```js
const { id, colorScheme, themeLight, themeDark } = options;
const { shadow, host } = createShadowHost({
  id,
  colorScheme,
  themeLight,
  themeDark
});
```